pkg/istio-agent/sds-agent.go

```diff
@@ -405,6 +393,10 @@ func (sa *Agent) newWorkloadSecretCache() (workloadSecretCache *cache.SecretCach
 		}
 	}
 
+	if sa.secOpts.TokenExchangers == nil {
```

Comment:
Can you add a comment ? Not sure why it has to be after the block - and not before as it was, so if there is a bug it would be good for the code to indicate why the ordering is important.